app/assets/javascripts/controllers/header/header_controller.js

```diff
@@ -0,0 +1,38 @@
+ManageIQ.angular.app.controller('headerController', HeaderCtrl);
+
+HeaderCtrl.$inject = ['$scope', 'eventNotifications'];
+
+function HeaderCtrl($scope, eventNotifications) {
+  var vm = this;
+
+  var cookieId = 'miq-notification-drawer';
+  vm.newNotifications = false;
+  vm.notificationsDrawerShown = sessionStorage.getItem(cookieId + "-shown") == 'true';
+  eventNotifications.setDrawerShown(vm.notificationsDrawerShown);
+
+  vm.toggleNotificationsList = function () {
+    vm.notificationsDrawerShown = !vm.notificationsDrawerShown;
+    sessionStorage.setItem(cookieId + "-shown", vm.notificationsDrawerShown);
+    eventNotifications.setDrawerShown(vm.notificationsDrawerShown);
+  };
+
+  var refresh = function() {
+    vm.newNotifications = eventNotifications.state().unreadNotifications;
+    try {
```

Comment:
Not so fond of this `$digest + catch` idiom, why is it even needed here?

Otherwise, maybe wrapping in `$timeout` is a bit more legible..?